docs/oathkeeper/pipeline/mutator.md

```diff
@@ -390,3 +392,106 @@ values out of an abundance of caution and for consistency.
   ]
 }
 ```
+
+## `hydrator`
+
+This mutator allows for fetching additional data from external APIs, which can
+be then used by other mutators. It works by making an upstream HTTP call to an
+API specified in the **Per-Rule Configuration** section below. The request is a
+POST request and it contains JSON representation of
+[AuthenticationSession](https://github.com/ory/oathkeeper/blob/d21179dd25543662075be402f6e24e1ee20d2754/pipeline/authn/authenticator.go#L26)
+struct in body, which is:
+
+```json
+{
+  "subject": String,
+  "extra": Object,
+  "header": Object
+}
+```
+
+As a response the mutator expects similiar JSON object, but with `extra` or
+`headers` fields modified. The AuthenticationSession from this object replaces
```

Comment:
is it `headers` or `header`? Maybe show an example response payload?

Reply:
This should be `header` of course 😅 
Regarding example response payload - I will provide it along with the fix of `headers`/`header`

Reply:
Just a small reminder that this still needs to be fixed :)

Reply:
@aeneasr can you take a look now?